pkg/registry/generic/etcd/etcd.go

```diff
@@ -459,16 +462,56 @@ func (e *Etcd) DeleteCollection(ctx api.Context, options *api.DeleteOptions, lis
 	if err != nil {
 		return nil, err
 	}
-	for _, item := range items {
-		accessor, err := meta.Accessor(item)
-		if err != nil {
-			return nil, err
-		}
-		if _, err := e.Delete(ctx, accessor.GetName(), options); err != nil {
-			return nil, err
+	// Spawn a number of goroutines, so that we can issue requests to etcd
+	// in parallel to speed up deletion.
+	workersNumber := registryoptions.GetDeleteCollectionParallelism()
+	wg := sync.WaitGroup{}
+	toProcess := make(chan int, 2*workersNumber)
```

Comment:
side note: we should probably create a util for scatter+gather, because we do it in a couple of locations.